File: G_Projects/F_Elevator_System/UtilityClasses/Building.java
package G_Projects.F_Elevator_System.UtilityClasses;

public class Building {
    private String name;
    private int numberOfFloors;
    private ElevatorController elevatorController; //Controller to manage elevators in the building

    public Building(String name, int numberOfFloors, int numberOfElevators){ 
        this.name = name;
        this.numberOfFloors = numberOfFloors;
        this.elevatorController = new ElevatorController(numberOfElevators, numberOfFloors);
    }

    public String getName(){
        return name;
    }
    public int getNumberOfFloors(){
        return numberOfFloors;
    }
    public ElevatorController getElevatorController(){
        return elevatorController;
    }
}

File: Python/A_Project/C_Elevator_System/UtilityClasses/building.py
from UtilityClasses.elevator_controller import ElevatorController


class Building:
    def __init__(self, name: str, number_of_floors: int, number_of_elevators: int):
        self.name = name
        self.number_of_floors = number_of_floors
        self.elevator_controller = ElevatorController(number_of_elevators, number_of_floors)

    def get_name(self) -> str:
        return self.name

    def get_number_of_floors(self) -> int:
        return self.number_of_floors

    def get_elevator_controller(self):
        return self.elevator_controller
